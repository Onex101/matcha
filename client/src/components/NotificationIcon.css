#notification-icon {
  display: inline-block;
}
#badge {
  position: absolute;
    top: 1px;
    /* left: -0.2px; */
    padding: 2px 7px;
    background-color: #4d7af5;
    color: white;
    font-size: 0.65em;
    border-radius: 50%;
    box-shadow: 1px 1px 1px gray;
    display: inherit;
    margin: 2px 15px;
}