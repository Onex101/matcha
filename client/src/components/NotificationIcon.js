import React, {Component} from "react";
import "./NotificationIcon.css";

class NotificationIcon extends Component {
    constructor(props) {
        super(props);
        this.state = { 
            count: 0
         }
    }

    componentDidUpdate() {
        // if (this.props.count != null) {
        //     console.log("COUNT 1")
        //     console.log(this.props.count)
        // }
        if (this.state.count !== this.props.count) {
            this.setState({count: this.props.count})
            console.log("COUNT")
            console.log(this.state.count)
        }
    }

    componentDidMount() {
        if (this.props.count != null ) {
            this.setState({count: this.props.count})
            console.log("COUNT")
            console.log(this.props.count)
        }
    }
    render() { 
        // this.setState({count: this.props.count})
        // if (this.state.count <= 0) {
        // if (this.state.count == null) {
        //     return (<div></div>)
        // }
        // console.log("ICON COUNT : " + this.props.count)

        return ( 
            <div id="notification-icon">
                <div id="badge">
                    {this.state.count}
                </div>
            </div>
         );
    }
}
 
export default NotificationIcon;