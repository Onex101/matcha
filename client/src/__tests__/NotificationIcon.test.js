import React from 'react';
import {render, cleanup,} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import NotificationIcon from '../components/NotificationIcon.js';

afterEach(cleanup);

describe("Testing Notification Icon", ()=>{
	it(" Display Notifications", ()=>{
		const { asFragment, getByTestId} = render(<NotificationIcon count={1} />);
		const badgeNode = getByTestId('badge');
		expect(badgeNode.textContent).toBe('1');
		expect(asFragment()).toMatchSnapshot();
	});
})
