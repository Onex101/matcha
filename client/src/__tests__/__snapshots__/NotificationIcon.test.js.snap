// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Notification Icon  Display Notifications 1`] = `
<DocumentFragment>
  <div
    id="notification-icon"
  >
    <div
      data-testid="badge"
      id="badge"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;
